SC_redesign/Crop_and_Soil/soil/carbon_cycling/residue_partition.py

```diff
@@ -305,3 +305,281 @@ def _determine_plant_structural_carbon_amount(plant_dry_matter_residue_amount: f
+
+        Parameters
+        ----------
+        weighted_residue_dry_matter_lignin_fraction: float
+            the weighted fractional of lignin amount in residue dry matter (unitless)
+        rainfall: float
+            amount of rain (mm H2O)
+
+        Returns
+        -------
+        float
+            the fraction of soil residue that's comprised of lignin
+
+        References
+        -------
+        pseudocode_soil S.6.B.II.3
+        """
+        return max(0.0, weighted_residue_dry_matter_lignin_fraction - 0.15 * rainfall * 0.01)
+
+    @staticmethod
+    def _determine_soil_lignin_to_nitrogen_ratio(plant_lignin_nitrogen_ratio: float,
+                                                 weighted_residue_dry_matter_lignin_fraction: float,
+                                                 soil_residue_lignin_fraction: float,
+                                                 nitrogen_fraction_plant_residue=0.4) -> float:
+        """This method calculates the soil lignin to nitrogen fraction
+
+        Parameters
+        ----------
+        plant_lignin_nitrogen_ratio: float
+            plant lignin to nitrogen ratio (dimensionless unit)
+        weighted_residue_dry_matter_lignin_fraction: float
+            weighted fraction of lignin in residue dry matter (unitless)
+        soil_residue_lignin_fraction: float
+            soil residue fraction that is comprised of lignin (unitless)
+        nitrogen_fraction_plant_residue: float default = 0.4
+            nitrogen fraction in plant residue at harvest (unitless)
+
+        Returns
+        -------
+        float
+            soil lignin to nitrogen fraction(unitless)
+
+        References
+        -------
+        pseudocode_soil S.6.B.II.4
+        """
+        if 0 < nitrogen_fraction_plant_residue <= 1:
+            return plant_lignin_nitrogen_ratio * weighted_residue_dry_matter_lignin_fraction + \
+                   (((soil_residue_lignin_fraction / 100) / nitrogen_fraction_plant_residue) / 100) \
+                   * (1 - weighted_residue_dry_matter_lignin_fraction)
+        elif nitrogen_fraction_plant_residue == 0:
+            return 0
+        else:
+            raise ValueError("Expected nitrogen_fraction_plant_residue be between 0.0-1.0, received "
+                             + str(nitrogen_fraction_plant_residue))
+
+    @staticmethod
+    def _determine_soil_residue_metabolic_fraction(soil_lignin_to_nitrogen_ratio: float) -> float:
+        """This method calculates the fraction of soil residue that is metabolic
+
+        Parameters
+        ----------
+        soil_lignin_to_nitrogen_ratio: float
+            soil lignin to nitrogen fraction(unitless)
+
+        Returns
+        -------
+        float
+            the fraction of soil residue that is metabolic(unitless)
+
+        References
+        -------
+        pseudocode_soil S.6.B.II.5
+        """
+        return 0.85 - 0.18 * soil_lignin_to_nitrogen_ratio
```

Comment:
```suggestion
        return max(0.0, 0.85 - 0.18 * soil_lignin_to_nitrogen_ratio)
```
I think it would make sense to lower bound this at 0